configurator/configurator-impl/src/main/java/org/codice/ddf/admin/configurator/impl/AdminOperation.java

```diff
@@ -92,23 +103,22 @@ public Void commit() throws ConfiguratorException {
                     pid));
         }
 
-        return null;
+        return ResultImpl.pass();
     }
 
     @Override
-    public Void rollback() throws ConfiguratorException {
+    public Result<Void> rollback() throws ConfiguratorException {
         try {
             saveConfigs(currentProperties);
         } catch (IOException | MalformedObjectNameException e) {
             throw new ConfiguratorException(String.format("Error rolling back configuration for %s",
                     pid));
         }
 
-        return null;
+        return ResultImpl.rollback();
     }
 
-    @Override
-    public Map<String, Object> readState() throws ConfiguratorException {
+    Map<String, Object> readState() throws ConfiguratorException {
         try {
             Map<String, Object> configResults = getConfigAdminMBean().getProperties(pid);
             if (configResults.isEmpty()) {
```

Comment:
We should really fix this while we're here. If a configuration is persisted and the blueprint default value is used, it does not show up as a service property. As a work around, we check the metatype for the default value which may differ from blueprint. There's also a bug where if only a portion of the configuration persisted used default values, then the metatype is never checked, making it appear that the service doesn't have the properties.